dvc/repo/__init__.py

```diff
@@ -80,7 +80,16 @@ def __init__(self, root_dir=None):
 
         self.config = Config(self.dvc_dir)
 
-        self.scm = SCM(self.root_dir)
+        if self.config.config.get(Config.SECTION_CORE, {}).get(
+            Config.SECTION_CORE_SUBREPO, False
+        ):
+            from dvc.repo.init import find_scm_dir
+
+            scm_root_dir = find_scm_dir(self.root_dir)
+        else:
+            scm_root_dir = self.root_dir
+
+        self.scm = SCM(scm_root_dir)
```

Comment:
Maybe we could pass `subrepo` or some flag to SCM and let it find the scm root? Feels like you are doing scm-related logic here for no good reason.